src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,167 @@
+package duke.command;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import duke.exception.DukeException;
+import duke.function.Storage;
+import duke.function.TaskList;
+import duke.function.Ui;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.ToDoTask;
+
+/**
+ * Represents a command to add a new task.
+ */
+public class AddCommand extends Command {
+    /**
+     * Type of task. Can be 'todo', 'deadline', or 'event
+     */
+    private String taskType;
+    /**
+     * The description and name of the task.
+     */
+    private String taskName;
+    /**
+     * The preposition for the date time. Only applicable for deadlines and events.
+     */
+    private String taskPreposition;
+    /**
+     * The date time for the event. Only applicable for deadlines and events.
+     */
+    private LocalDateTime taskDate;
+    /**
+     * To store any exceptions that were thrown during the parsing of the command.
+     */
+    private DukeException exception;
```

Comment:
I really like how thorough you are with these JavaDoc comments. 
Although, for completeness, I think the first comment (line 23) might be missing a few characters.
Shouldn't `...event` be `...event'.` ?